src/equations/shallow_water_exner_1d.jl

```diff
@@ -0,0 +1,618 @@
+
+    # Eigenvalue absolute value matrix
+    Λ_abs = @SMatrix [abs(λ1) z z; z abs(λ2) z; z z abs(λ3)]
+
+    # Compute dissipation
+    diss = SVector(-0.5 * R * Λ_abs * R_inv * (u_rr - u_ll))
+
+    return SVector(diss[1], diss[2], diss[3])
+end
+
+@inline function Trixi.max_abs_speed_naive(u_ll, u_rr, orientation::Integer,
+                                           equations::ShallowWaterExnerEquations1D)
+    return max(eigvals_cardano(u_rr, equations)..., eigvals_cardano(u_ll, equations)...)
+end
+
+@inline function Trixi.max_abs_speeds(u, equations::ShallowWaterExnerEquations1D)
+    return maximum(eigvals_cardano(u, equations))
+end
+
+#Helper function to extract the velocity vector from the conservative variables
+@inline function Trixi.velocity(u, equations::ShallowWaterExnerEquations1D)
+    h, hv, _ = u
+
+    v = hv / h
+
+    return v
+end
+
+# Compute the sediment discharge for Shields stress models
+@inline function q_s(u,
+                     equations::ShallowWaterExnerEquations1D{T, S,
+                                                             ShieldsStressModel{T}}) where {
+                                                                                            T,
+                                                                                            S
+                                                                                            }
+    (; gravity, rho_f, rho_s, porosity_inv) = equations
+    (; m_1, m_2, m_3, k_1, k_2, k_3, theta_c, d_s) = equations.sediment_model
+
+    theta = rho_f * abs(shear_stress(u, equations)) / (gravity * (rho_s - rho_f) * d_s)  # Shields stress
+
+    Q = d_s * sqrt(gravity * (rho_s / rho_f - 1.0) * d_s) # Characteristic discharge
+
+    return porosity_inv * Q * sign(theta) * k_1 * theta^m_1 *
+           (max(theta - k_2 * theta_c, 0.0))^m_2 *
+           (max(sqrt(theta) - k_3 * sqrt(theta_c), 0.0))^m_3
+end
+
+# Compute the sediment discharge for the Grass model 
+@inline function q_s(u,
+                     equations::ShallowWaterExnerEquations1D{T, S, GrassModel{T}}) where {
+                                                                                          T,
+                                                                                          S
+                                                                                          }
+    (; porosity_inv, sediment_model) = equations
+    return porosity_inv * sediment_model.A_g * velocity(u, equations)^3
+end
+
+# Shear stress formulation using a coefficient to take into account different friction models
+@inline function shear_stress(u, equations::ShallowWaterExnerEquations1D)
+    v = velocity(u, equations)
+    return equations.gravity * shear_stress_coefficient(u, equations.friction) * v *
+           abs(v)
+end
+
+# Model dependent shear stress coefficient
+@inline function shear_stress_coefficient(u, friction::ManningFriction)
+    h, _, _ = u
+    return friction.n^2 / h^(1 / 3)
```

Comment:
To handle wetting/drying this would need some kind of designularization, right?

Reply:
Yes, near dry states we would need to add some kind of desingularization here to deal with the vanishing `h`.